### DISCUSSION TITLE
Setting residual phase in Imbibition algorithm

### DISCUSSION THREAD

**Discussion by kizzle4shaw** [PMEAL/OpenPNM#2852] [Q&A]
Hello, 
I just wanted to know, will there be an algorithm like mixed invasion percolation in this version of openpnm? 
This is because I'm having difficulties in using the old version, most functionalities don't work for some reason. 

Also the setting residual saturation fuctionality as seen from the source code gives an error, will it be available in this version?

I just want to know how much the tools can be used after network extraction especially the simulation part.

**Comment by ma-sadeghi** [comment]
We're bandwidth strapped. If you're familiar with the inner workings of OpenPNM, you should be able to port the old version to work with v3, and possibly be merged in as a pull request. If you're willing, once you have the initial draft, you can ping us in the PR and we can guide you along the way. That's the best we can do at the moment.

**Reply by kizzle4shaw** [reply]
I don't quite understand. Like I should fork both versions and then change the mixed invasion percolation arguments to v3 requirements? then merge with v3? Please if you have time, you can walk me through the steps so I can be able to use the old version's simulation capabilities especially after extracting the network and checking it's health.
This is because the old version offers more when imbibition and trapping is concerned. 
Hopefully you'll help me on that.

**Reply by ma-sadeghi** [reply]
Not quite. The old version cannot be merged into v3, so you'd have to manually pick the files needed to run the old code and add them to v3. Of course it won't work, but this is where you'd start. From there, it's just an iterative process until you get it to run. Let me know if that makes sense

**Reply by kizzle4shaw** [reply]
Yes it makes sense. I think it requires expert python knowledge. The algorithms are classes and most of them inherit from somewhere which means in v3 it will be very difficult to use them, by now the differences are too much. I will try as much as I can.

**Reply by ma-sadeghi** [reply]
Try using chatgpt especially if you need help with established python knowledge (which is the case here). You might even be able to get it to write most of the boilerplate for you
